### ISSUE TITLE
Location does not updated continuously

### ISSUE THREAD

**Issue by kalisoft93** [OneSignal/OneSignal-iOS-SDK#815]
**Description:**

Hello,

We have an issue with continuous location sharing also when the app running on background or foreground.
Our problem is that the location points update only once when we launch the app, not updating location continuously in foreground neither in background, so we need to kill the app and relaunch it for update location points again only once.

Every permission has added to Info.plist what you mentioned in OneSignal SDK Doc:

```
<string>It needs your location</string>
<key>NSLocationAlwaysUsageDescription</key>
<string>It requires location</string>
<key>NSLocationUsageDescription</key>
<string>Your message goes here</string>
<key>NSLocationWhenInUseUsageDescription</key>
<string>Allow location tracking</string>
<key>NSMicrophoneUsageDescription</key>
```

Also we checked Background Modes:
```
<key>UIBackgroundModes</key>
	<array>
		<string>location</string>
		<string>remote-notification</string>
	</array>
```
We call promptLocation before initWithLaunchOptions

Also we have observed the promptLocation open only Allow while Using App/ Allow once dialog for location permission, and not opening Always allow dialog.

**Environment**

 IOS SDK 14.* 
OneSignal release 2.16

**Comment by mbx005** [comment]
We have the same issue! Is anybody has a solution to resolve this issue?

**Comment by emawby** [comment]
Hello, thank you for the report. The foreground piece of this issue is being fixed in #856 and should be available with the next release. 
The background piece of the issue has also been reproduced and a path forward is being investigated.
